sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/ExpressionEncoder.scala

```diff
@@ -43,10 +44,11 @@ import org.apache.spark.util.Utils
  *    to the name `value`.
  */
 object ExpressionEncoder {
+
   def apply[T : TypeTag](): ExpressionEncoder[T] = {
-    // We convert the not-serializable TypeTag into StructType and ClassTag.
     val mirror = ScalaReflection.mirror
-    val tpe = typeTag[T].in(mirror).tpe
+    val tpe = ScalaReflection.localTypeOf[T]
```

Comment:
why change it from `typeTag[T].in(mirror).tpe`?

Reply:
`localTypeOf` is actually doing the same thing. I think it is better to use ScalaReflection for such thing.

Reply:
I think it should be fine, but let me revert this change first.